PlaceRouteHierFlow/cap_placer/capplacer.cpp

```diff
@@ -1440,574 +1325,396 @@ void Placer_Router_Cap::Router_Cap(vector<int> & ki, vector<pair<string, string>
+                      n.start_conection_coord.push_back(coord);
+                      coord.first = Caps[n.cap_index[k]].x- unit_cap_demension.first/2-(span_distance.first-min_dis_x*trails[l]);
+                      Caps[n.cap_index[k]].access = 1;
             
-                      Nets_pos[i].end_conection_coord.push_back(coord);
-                      Nets_pos[i].Is_pin.push_back(0);
-                      Nets_pos[i].metal.push_back(H_metal);
+                      n.end_conection_coord.push_back(coord);
+                      n.Is_pin.push_back(0);
+                      n.metal.push_back(H_metal);
 
-                      addVia(Nets_pos[i],coord,drc_info,HV_via_metal,HV_via_metal_index,0);                     
-                      //
-                      //Nets_neg[i].line_v[l]=0;
-                      //Nets_neg[i].line_v[l+1]=1;
-                      
-                      //
-                      
-                      //
-                      if(Caps[Nets_pos[i].cap_index[k]].index_y>=max){
-                         max=Caps[Nets_pos[i].cap_index[k]].index_y;
-                         max_cap_index = Nets_pos[i].cap_index[k];
-                         left_right = 0;
-                        }
-                    }else if(l-Caps[Nets_pos[i].cap_index[k]].index_x==1 and Caps[Nets_pos[i].cap_index[k]].access==0){
+                      addVia(n,coord,drc_info,HV_via_metal,HV_via_metal_index,0);                     
+		      mb.update( Caps[n.cap_index[k]].index_y, n.cap_index[k], 0);
+
+                    }else if(l-Caps[n.cap_index[k]].index_x==1 and Caps[n.cap_index[k]].access==0){
                       found = 1;
-                      coord.first = Caps[Nets_pos[i].cap_index[k]].x+ unit_cap_demension.first/2-shifting_x;
-                      coord.second = Caps[Nets_pos[i].cap_index[k]].y- unit_cap_demension.second/2+shifting_y;
+                      coord.first = Caps[n.cap_index[k]].x+ sign*(unit_cap_demension.first/2-shifting_x);
+                      coord.second = Caps[n.cap_index[k]].y- sign*(unit_cap_demension.second/2-shifting_y);
                       
                       //
-                      addVia(Nets_pos[i],coord,drc_info,HV_via_metal,HV_via_metal_index,0);
+                      addVia(n,coord,drc_info,HV_via_metal,HV_via_metal_index,0);
                     
-                      Nets_pos[i].start_conection_coord.push_back(coord);
-                      coord.second = Caps[Nets_pos[i].cap_index[k]].y- unit_cap_demension.second/2+shifting_y-min_dis_y;
-                      Nets_pos[i].end_conection_coord.push_back(coord);
-                      Nets_pos[i].Is_pin.push_back(0);
-                      Nets_pos[i].metal.push_back(V_metal);
-
-                      addVia(Nets_pos[i],coord,drc_info,HV_via_metal,HV_via_metal_index,0);
+                      n.start_conection_coord.push_back(coord);
+		      if ( sign == 1) { // SMB This really looks wrong
```

Comment:
@Lastdayends 
Here is an obvious issue with the code after merging the code for pos_net and neg_net into one common routine.